haskell-src/exec/Chainweb/Server.hs

```diff
@@ -433,9 +433,10 @@ queryTxsByKey logger rk c =
         , Api._signer_capList = caps
         }
     let sigs = Api.Sig . unSignature . _signer_sig <$> dbSigners
+        sameBlock tx ev = (unBlockId $ _tx_block tx) == (unBlockId $ _ev_block ev)
```

Comment:
I feel like you should put the wanted events in the let binding. Like this:

```
evsInTxBlock = filter (sameBlock tx) evs
```

Reply:
Thanks for the suggestion, that `toApiTxDetail` call was indeed busy enough already so I've applied your suggestion.